src/AppBundle/Security/AccountVoter.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace AppBundle\Security;
+
+use AppBundle\Entity\Account;
+use AppBundle\Entity\Profile;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Authorization\Voter\Voter;
+
+class AccountVoter extends Voter
+{
+    const ACCESS = 'access';
+
+    protected function supports($attribute, $subject)
+    {
+        if ($attribute !== self::ACCESS) {
+            return false;
+        }
+
+        if (!$subject instanceof Account) {
+            return false;
+        }
+
+        return true;
+    }
+
+    protected function voteOnAttribute($attribute, $subject, TokenInterface $token)
+    {
+        $user = $token->getUser();
+
+        if (!$user instanceof Profile) {
```

Comment:
Maybe we should rename profile to `UserProfile` to get things easier to understand.